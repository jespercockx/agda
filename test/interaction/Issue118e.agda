
data _≡_ {A : Set} (x : A) : A → Set where
  refl : x ≡ x

data A : Set where
  a : A

mutual

  data S : Set where
    c : (s : S) (x : A) → f s ≡ x → S

  f : S → A
  f (c _ x _) = x

{-# TERMINATING #-}
foo : S
foo = c foo a {!refl!}
