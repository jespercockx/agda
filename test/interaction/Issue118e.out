(agda2-status-action "")
(agda2-info-action "*Type-checking*" "" nil)
(agda2-highlight-clear)
(agda2-status-action "")
(agda2-info-action "*All Goals*" "?0 : f foo ≡ a " nil)
((last . 1) . (agda2-goals-action '(0)))
(agda2-info-action "*Error*" "1,1-5 (f foo) != a of type A when checking that the expression refl has type f foo ≡ a" nil)
(agda2-highlight-load-and-delete-action)
(agda2-status-action "")
